pages/api/gql/[domain]/stats.js

```diff
@@ -0,0 +1,197 @@
+import { parse } from 'cookie';
+import { differenceInDays, differenceInMinutes, sub } from 'date-fns';
+
+import { ok } from 'lib/response';
+import { getGQLUrl } from 'utils/urls';
+// eslint-disable-next-line import/no-anonymous-default-export
+export default async (req, res) => {
+  const jwtToken = parse(req.headers.cookie || '')['jwt'];
+  const { start_at, end_at, domain, groupByUnit, groupByValue, url, ref } = req.query;
+  const data = await getAnalyticsData({
+    token: jwtToken,
+    domain,
+    startDate: start_at,
+    endDate: end_at,
+    groupByUnit,
+    groupByValue,
+    url,
+    ref,
+  });
+  return ok(res, data);
+};
+
+async function getAnalyticsData({ token, domain, startDate, endDate, groupByValue, url, ref }) {
+  try {
+    const from = new Date(parseInt(startDate)),
+      to = new Date(parseInt(endDate));
+    const differenceKeyValuePair = getDifferenceKeyValuePair(groupByValue, from, to);
+    const pastFrom = sub(from, differenceKeyValuePair);
```

Comment:
When selecting the "this year" time than this is an invalid date. I have added some logs showing this:
![image](https://github.com/Hashnode/umami/assets/7524745/86257cbe-7efe-457f-baab-16799084c80f)

This may cause the comparison to the previous time period in this screenshot to be off:
![image](https://github.com/Hashnode/umami/assets/7524745/b3929b63-632b-49f4-bced-e7015feda174)

Reply:
Fixed it myself since I have found the reason quickly.